### PR TITLE
Dat 34822 fix input

### PR DESCRIPTION
**Pull request by YonathanMeguira**
Fixes for inputs issues

### CODE REVIEW

File: lib/src/input/input.component.scss [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 :host .dato-icon-close {
   cursor: pointer;
-  z-index: 1002;
+  z-index: 10;
```
Comment (bh86) [inline lib/src/input/input.component.scss]: please remove the `z-index` property

File: lib/src/input/input.component.ts [hunk 2]
```diff
@@ -79,10 +78,18 @@ export class DatoInputComponent extends BaseCustomControl implements OnInit, OnD
   /**
    * Get the native input
    */
-  get inpuElement(): HTMLInputElement {
+  get inputElement(): HTMLInputElement {
```
Comment (bh86) [inline lib/src/input/input.component.ts]: great typo fix! But please note that this change will cause a breaking change.
So please keep both getters (old & new) and add a deprecated JSDoc above the old getter.

File: lib/src/input/input.component.ts [hunk 3]
```diff
@@ -79,10 +78,18 @@ export class DatoInputComponent extends BaseCustomControl implements OnInit, OnD
...
     return this.host.nativeElement.querySelector('input');
   }
 
+  /**
+   * get the close icon
+   */
+
```
Comment (bh86) [inline lib/src/input/input.component.ts]: redundant newline

File: lib/src/input/input.component.ts [hunk 4]
```diff
@@ -122,11 +129,12 @@ export class DatoInputComponent extends BaseCustomControl implements OnInit, OnD
    * @param {boolean} isDisabled
    */
   setDisabledState(isDisabled: boolean): void {
-    this.renderer.setProperty(this.inpuElement, 'disabled', isDisabled);
+    this.renderer.setProperty(this.inputElement, 'disabled', isDisabled);
+    this.renderer.setStyle(this.closeIcon, 'pointer-events', 'none');
```
Comment (bh86) [inline lib/src/input/input.component.ts]: You're always set the 'pointer-events' to `none`, regardless of the value of `isDisabled`.
better to toggle the `disabled` attribute (as above) and add a new CSS selector for the disabled attribute.

File: lib/src/input/input.component.scss (hunk 1 above)
Comment (YonathanMeguira) [reply lib/src/input/input.component.scss]: done

File: lib/src/input/input.component.ts (hunk 3 above)
Comment (YonathanMeguira) [reply lib/src/input/input.component.ts]: fixed

File: lib/src/input/input.component.ts (hunk 4 above)
Comment (YonathanMeguira) [reply lib/src/input/input.component.ts]: fixed

File: lib/src/input/input.component.ts (hunk 2 above)
Comment (YonathanMeguira) [reply lib/src/input/input.component.ts]: fixed

**Pull request merged**